scripts/build-packages-dist.mts

```diff
@@ -0,0 +1,155 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+/**
+ * Script that builds the release output of all packages which have the "release-package
+ * Bazel tag set. The script builds all those packages and copies the release output to the
+ * distribution folder within the project.
+ */
+
+import { BuiltPackage } from '@angular/ng-dev';
+import { execSync } from 'node:child_process';
+import { copyFileSync, chmodSync, mkdirSync, rmSync } from 'node:fs';
+import { dirname, join } from 'node:path';
+import { fileURLToPath } from 'node:url';
+import sh from 'shelljs';
+
+/** Name of the Bazel tag that will be used to find release package targets. */
+const releaseTargetTag = 'release-package';
+
+/** Path to the project directory. */
+const projectDir = join(dirname(fileURLToPath(import.meta.url)), '../');
+
+/** Command that runs Bazel. */
+const bazelCmd = process.env.BAZEL || `yarn -s bazel`;
+
+/** Command that queries Bazel for all release package targets. */
+const queryPackagesCmd =
+  `${bazelCmd} query --output=label "attr('tags', '\\[.*${releaseTargetTag}.*\\]', //packages/...) ` +
+  `intersect kind('pkg_npm', //packages/...)"`;
+
+/** Path for the default distribution output directory. */
+const defaultDistPath = join(projectDir, 'dist/releases');
+
+/** Builds the release packages for NPM. */
+export function performNpmReleaseBuild(): BuiltPackage[] {
+  return buildReleasePackages(defaultDistPath, /* isSnapshotBuild */ false);
+}
+
+/**
+ * Builds the release packages as snapshot build. This means that the current
+ * Git HEAD SHA is included in the version (for easier debugging and back tracing).
+ */
+export function performDefaultSnapshotBuild(): BuiltPackage[] {
+  return buildReleasePackages(defaultDistPath, /* isSnapshotBuild */ true);
+}
+
+/**
+ * Builds the release packages with the given compile mode and copies
+ * the package output into the given directory.
+ */
+function buildReleasePackages(distPath: string, isSnapshotBuild: boolean): BuiltPackage[] {
+  console.log('######################################');
+  console.log('  Building release packages...');
+  console.log('######################################');
+
+  // List of targets to build. e.g. "packages/angular/cli:npm_package"
+  const targets = exec(queryPackagesCmd, true)
+    .split(/\r?\n/)
+    .map((target) => target + '_archive');
+  const packageNames = getPackageNamesOfTargets(targets);
+  const bazelBinPath = exec(`${bazelCmd} info bazel-bin`, true);
+  const getBazelOutputPath = (pkgName: string) =>
+    join(bazelBinPath, 'packages', pkgName, 'npm_package');
```

Comment:
I think this would just directly point to the archive right? It's a little odd because you are only requesting the archives, but also copy the `npm_package` tree artifact/folder too. Maybe both targets should be explicitly requested and we have two functions like `getOutputNpmPackagePath()` and `getOutputNpmPackageArchivePath()`

Reply:
Updated to build both targets and added separate variables for each output type (directory/archive).
I'd like to move towards just using the generated archives in the future to reduce the amount of files/directories that need to be copied and managed.

Reply:
Yup, agreed!